sui-execution/latest/sui-adapter/src/programmable_transactions/trace_utils.rs

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::execution_value::{ObjectContents, ObjectValue, Value};
+use crate::programmable_transactions::context::*;
+use move_core_types::language_storage::StructTag;
+use move_core_types::{identifier::Identifier, language_storage::TypeTag};
+use move_trace_format::{
+    format::{MoveTraceBuilder, RefType, TraceEvent, TypeTagWithRefs},
+    value::{SerializableMoveValue, SimplifiedMoveStruct},
+};
+use move_vm_types::loaded_data::runtime_types::Type;
+use sui_types::coin::Coin;
+use sui_types::object::bounded_visitor::BoundedVisitor;
+use sui_types::ptb_trace::{PTBCommandInfo, PTBExternalEvent, SplitCoinsEvent, TransferEvent};
+use sui_types::transaction::Command;
+use sui_types::{
+    base_types::ObjectID,
+    error::{ExecutionError, ExecutionErrorKind},
+    ptb_trace::MoveValueInfo,
+};
+
+/// Inserts Move call start event into the trace.
+pub fn trace_move_call_start(trace_builder_opt: &mut Option<MoveTraceBuilder>) {
+    if let Some(trace_builder) = trace_builder_opt {
```

Comment:
All public functions are in this file are guarded by an `if` statement containing the whole function body that ensures any code only being executed when tracing is enabled. The only entry point to the execution layer that enables tracing is in the new replay tool [here](https://github.com/MystenLabs/sui/blob/0886cb100d7567a4ee6c8a121c6d688ee7c2715f/crates/sui-replay-2/src/execution.rs#L93)